src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Parser.cs

```diff
@@ -812,9 +837,26 @@ private DictionarySpec CreateDictionarySpec(INamedTypeSymbol type, Location? loc
                                     RegisterHasChildrenHelperForGenIfRequired(propertyTypeSpec);
                                 }
 
+                                PropertySpec spec = new(property) { Type = propertyTypeSpec, ConfigurationKeyName = configKeyName };
+
+                                ParameterSpec? matchingParam = null;
+                                if ((parameterCache?.TryGetValue(propertyName, out matchingParam)) is true &&
```

Comment:
Is case sensitive association of properties to parameters something that the reflection binder is doing? Or is it just using the convention we follow in STJ?

Reply:
Do you mean case insensitive? If so, yes the reflection binder does this.

Reply:
Yes I meant case insensitive, apologies.